### DISCUSSION TITLE
blog/detect-page-navigation-in-next14

### DISCUSSION THREAD

**Discussion by giscus** [yunjeoming/yuns-blog#59] [Comments]
# blog/detect-page-navigation-in-next14

설정 화면에서 벗어날 때와 같은 페이지 이동을 어떻게 감지해야할까?

https://yunjeoming.dev/blog/detect-page-navigation-in-next14

**Comment by ohbyul** [comment]
안녕하세요 글 잘 봤습니다 isSaved 는 어떤 값을 담는 건지 알 수 있을까요?

**Reply by ryu9663** [reply]
저도 궁금합니다!

**Reply by JaeHyeok-Han** [reply]
특정 폼이 작성되는 도중에 창을 벗어나는 것을 막기 위해 해당 훅을 만드신 것을 봐서는 폼이 저장되었는지 여부를 나타내는 것 같습니다.
결국 커스텀 훅 방식으로 사용하게 되니 해당 훅을 사용할때 isSaved와 저장되지 않았을때 페이지를 벗어나는 것이 감지되었을때 실행할 함수(모달 띄우기 등)를 넘겨서 실행하지 않을까요??
아마 작성자분께서 코드 전문을 올리진 않으신걸로 보이네요 :)
